### DISCUSSION TITLE
Calculatedvalue with objectbrick getter not working after update to 10.4

### DISCUSSION THREAD

**Discussion by Hbitvof** [pimcore/pimcore#12119] [Q&A / Support]
In the release notes of 10.4 there is:
Important: [Object bricks] A call to the object brick´s getter method no longer returns object bricks marked for deletion. To restore the original behavior pass "true" to the getter method´s $includeDeletedBricks argument.

Now I have this problem, but I have no clue what to change to get this working again. Can you please provide an example because this is not enough information for me to get this working.

**Comment by mcop1** [comment]
Hello, 

for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:

```
$obj = DataObject::getById(81);
$attr = $obj->getAttributes();
$brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
$brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
//$brick->...
```

Hope this helps,
best regards

**Reply by Hbitvof** [reply]
> Hello,
> 
> for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:
> 
> ```
> $obj = DataObject::getById(81);
> $attr = $obj->getAttributes();
> $brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
> $brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
> //$brick->...
> ```
> 
> Hope this helps, best regards

I use the following code to check if the objectbrick is filled in: 

`$object->geteigenschappen()->getVorm($context->getPosition())`

In your example i read it's not needed to use true but I get this error:

Fatal error:  Uncaught TypeError: Pimcore\Model\DataObject\Product\Eigenschappen::getVorm(): Argument #1 ($includeDeletedBricks) must be of type bool, null given

This code works in Pimcore 6.9.x. What do I need to change here?

**Reply by mcop1** [reply]
> > Hello,
> > for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:
> > ```
> > $obj = DataObject::getById(81);
> > $attr = $obj->getAttributes();
> > $brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
> > $brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
> > //$brick->...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Hope this helps, best regards
> 
> I use the following code to check if the objectbrick is filled in:
> 
> `$object->geteigenschappen()->getVorm($context->getPosition())`
> 
> In your example i read it's not needed to use true but I get this error:
> 
> Fatal error: Uncaught TypeError: Pimcore\Model\DataObject\Product\Eigenschappen::getVorm(): Argument #1 ($includeDeletedBricks) must be of type bool, null given
> 
> This code works in Pimcore 6.9.x. What do I need to change here?

What are you trying to archieve by passing `$context->getPosition()` as an argument? Because before we introduced the `$includeDeletedBricks` argument, the method didn´t accept any arguments at all. Could you please check the signature of `getVorm` too?

**Reply by Hbitvof** [reply]
> > > Hello,
> > > for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:
> > > ```
> > > $obj = DataObject::getById(81);
> > > $attr = $obj->getAttributes();
> > > $brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
> > > $brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
> > > //$brick->...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Hope this helps, best regards
> > 
> > 
> > I use the following code to check if the objectbrick is filled in:
> > `$object->geteigenschappen()->getVorm($context->getPosition())`
> > In your example i read it's not needed to use true but I get this error:
> > Fatal error: Uncaught TypeError: Pimcore\Model\DataObject\Product\Eigenschappen::getVorm(): Argument #1 ($includeDeletedBricks) must be of type bool, null given
> > This code works in Pimcore 6.9.x. What do I need to change here?
> 
> What are you trying to archieve by passing `$context->getPosition()` as an argument? Because before we introduced the `$includeDeletedBricks` argument, the method didn´t accept any arguments at all. Could you please check the signature of `getVorm` too?

This is the only way that worked to check in the objectbrick exists on the object, every other check didn't work. In this case this is a select objectbrick type. For an input objectbrick type this is not needed.

I guess there is another/better way to check if this objectbrick exists on the object that solves this problem?

**Reply by mcop1** [reply]
> > > > Hello,
> > > > for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:
> > > > ```
> > > > $obj = DataObject::getById(81);
> > > > $attr = $obj->getAttributes();
> > > > $brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
> > > > $brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
> > > > //$brick->...
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Hope this helps, best regards
> > > 
> > > 
> > > I use the following code to check if the objectbrick is filled in:
> > > `$object->geteigenschappen()->getVorm($context->getPosition())`
> > > In your example i read it's not needed to use true but I get this error:
> > > Fatal error: Uncaught TypeError: Pimcore\Model\DataObject\Product\Eigenschappen::getVorm(): Argument #1 ($includeDeletedBricks) must be of type bool, null given
> > > This code works in Pimcore 6.9.x. What do I need to change here?
> > 
> > 
> > What are you trying to archieve by passing `$context->getPosition()` as an argument? Because before we introduced the `$includeDeletedBricks` argument, the method didn´t accept any arguments at all. Could you please check the signature of `getVorm` too?
> 
> This is the only way that worked to check in the objectbrick exists on the object, every other check didn't work. In this case this is a select objectbrick type. For an input objectbrick type this is not needed.
> 
> I guess there is another/better way to check if this objectbrick exists on the object that solves this problem?

If you want to check whether the brick was added or not, you can use something simple like this:
```
$brick = $attr->getBodywork();
  if ($brick) {
      //brick added
  } else {
      //brick not added
  }
```
Therefore your code should look something like this:
```
$brick = $object->geteigenschappen()->getVorm();
if ($brick) {
    //brick added
} else {
    //brick not added
}
```

**Reply by Hbitvof** [reply]
> > > > > Hello,
> > > > > for example, take a look at the Bodywork object brick in the "Car" data object class at http://demo.pimcore.fun. To get the brick, you would call the "getBodyWork" getter method and there you can pass "true" if you want to get object bricks marked as deleted:
> > > > > ```
> > > > > $obj = DataObject::getById(81);
> > > > > $attr = $obj->getAttributes();
> > > > > $brick = $attr->getBodywork(); //doesn´t include object bricks marked as deleted
> > > > > $brick = $attr->getBodywork(true); //this however does include object bricks marked as deleted
> > > > > //$brick->...
> > > > > ```
> > > > > 
> > > > > 
> > > > >     
> > > > >       
> > > > >     
> > > > > 
> > > > >       
> > > > >     
> > > > > 
> > > > >     
> > > > >   
> > > > > Hope this helps, best regards
> > > > 
> > > > 
> > > > I use the following code to check if the objectbrick is filled in:
> > > > `$object->geteigenschappen()->getVorm($context->getPosition())`
> > > > In your example i read it's not needed to use true but I get this error:
> > > > Fatal error: Uncaught TypeError: Pimcore\Model\DataObject\Product\Eigenschappen::getVorm(): Argument #1 ($includeDeletedBricks) must be of type bool, null given
> > > > This code works in Pimcore 6.9.x. What do I need to change here?
> > > 
> > > 
> > > What are you trying to archieve by passing `$context->getPosition()` as an argument? Because before we introduced the `$includeDeletedBricks` argument, the method didn´t accept any arguments at all. Could you please check the signature of `getVorm` too?
> > 
> > 
> > This is the only way that worked to check in the objectbrick exists on the object, every other check didn't work. In this case this is a select objectbrick type. For an input objectbrick type this is not needed.
> > I guess there is another/better way to check if this objectbrick exists on the object that solves this problem?
> 
> If you want to check whether the brick was added or not, you can use something simple like this:
> 
> ```
> $brick = $attr->getBodywork();
>   if ($brick) {
>       //brick added
>   } else {
>       //brick not added
>   }
> ```
> 
> Therefore your code should look something like this:
> 
> ```
> $brick = $object->geteigenschappen()->getVorm();
> if ($brick) {
>     //brick added
> } else {
>     //brick not added
> }
> ```

This worked, Thanks for your help!
